src/frontend/js/widgets/SyllabusCourseRunsList/components/CourseProductItem/index.tsx

```diff
@@ -64,7 +64,7 @@ const Header = ({ product, order, hasPurchased, isPendingState, compact }: Heade
   const formatDate = useDateFormat();
 
   const canShowMetadata = useMemo(() => {
-    return compact && !order;
+    return compact && (!order || [OrderState.SUBMITTED, OrderState.PENDING].includes(order!.state));
```

Comment:
By reading tests this seems wrong.
Tests expect to show metadata until order is `VALIDATED`

Reply:
Yes + a bit of comment to make easier the reading of this condition would be great too

Reply:
Before: we simply create an `Order` with `OrderFactory`
Now: we mock the api call made by `useCurrentOrder`, it add the need of `JoanieSession` and all these small changes.